authors/apps/articles/views.py

```diff
@@ -129,3 +143,110 @@ def destroy(self, request, *args, **kwargs):
         super().destroy(self, request, *args, **kwargs)
 
         return Response({'message': 'The article has been deleted.'})
+
+
+class CommentAPIView(ListCreateAPIView):
+    queryset = Comment.objects.all()
+    serializer_class = CommentSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly, )
+    """This class list and create comment"""
+
+    # filter by slug from url
+    lookup_url_kwarg = 'slug'
+    lookup_field = 'article__slug'
+
+    def filter_queryset(self, queryset):
+        """This method filter and get comment of an article."""
+        filters = {self.lookup_field: self.kwargs[self.lookup_url_kwarg]}
+        return queryset.filter(**filters)
+
+    def create(self, request, *args, **kwargs):
+        """This methods creates a comment"""
+        try:
+            article = Article.objects.get(slug=kwargs['slug'])
+        except Exception:
+            return Response({'Error,Please check your url?'},
```

Comment:
I don't think you should be handling this by yourself. The `url_pattern` should handle it,

Reply:
@esirK, do you mean the response ?`url_pattern`handles the slugs error but it's not in JSON format